portal/models/qb_timeline.py

```diff
@@ -0,0 +1,290 @@
+from datetime import datetime, MAXYEAR
+
+from ..database import db
+from ..date_tools import RelativeDelta
+from .assessment_status import OverallStatus
+from .questionnaire_bank import QuestionnaireBank, qbs_by_rp
+from .questionnaire_response import QNR_results
+from .user_consent import consent_withdrawal_dates
+
+
+class QBT(db.Model):
+    """Effectively a view, to simplify QB status lookups over time
+
+    A user has a number of QBT rows, at least one for each questionnaire
+    bank that has been due for the user, or will be in the future.
+
+    The following events would invalidate the respective rows (delete
+    said rows to invalidate):
+     - user's trigger date changed (consent or new procedure)
+     - user submits a QuestionnaireResponse
+     - the definition of a QB or an organization's research protocol
+
+    The table is populated up to the next known event for a user.  If a
+    future date isn't found, that user's data is due for update.
+
+    """
+    __tablename__ = 'questionnaire_bank_timeline'
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.ForeignKey(
+        'users.id', ondelete='cascade'), nullable=False)
+    at = db.Column(
+        db.DateTime, nullable=False,
+        doc="initial date time for state of row")
+    qb_id = db.Column(db.ForeignKey(
+        'questionnaire_banks.id', ondelete='cascade'), nullable=True)
+    qb_iteration = db.Column(db.Integer, nullable=True)
+    _status = db.Column('status', db.Text, nullable=False)
+
+    @property
+    def status(self):
+        return self._status
+
+    @status.setter
+    def status(self, value):
+        self._status = getattr(OverallStatus, value).name
+
+
+system_trigger = datetime(2015, 1, 1, 12, 0, 0)
```

Comment:
Maybe something further back and standardized, like the Unix Epoch?

Reply:
we ran into trouble with dates < 1900 w/ one of the python date libraries.  can go to that if you'd rather?  (jan 1, 1900)

Reply:
killed this approach after finding it dysfunctional w/ the "month math" we require.